cmd/restic/cmd_prune.go

```diff
@@ -316,7 +316,9 @@ func prune(opts PruneOptions, gopts GlobalOptions, repo restic.Repository, usedB
 			return nil
 		}
 
-		if p.unusedSize+p.usedSize != uint64(packSize) {
+		if p.unusedSize+p.usedSize != uint64(packSize) &&
+			(p.usedBlobs != 0 || p.duplicateBlobs != 0) {
```

Comment:
I'd prefer `!(p.usedBlobs == 0 && p.duplicateBlobs == 0)`. That way it's more obvious that this corresponds to the "remove pack"-case later on. Please also add a comment, which states that it is permitted to remove a damaged but unnecessary pack file. (The main difference to the current comment is that I'm asking for why a certain behavior is _correct_ and not whether a certain case is an error.)